### PR TITLE
Horoscope

### PR DESCRIPTION
**Pull request by gduck**
user selects a date. the ng-change function sends an ajax get to a url, which routes to my zodiac controller. The find method of the controller scrapes the data and sets up the variable for the json.jbuilder. On success, the page variables are populated with the json
It's a beautiful thing

### CODE REVIEW

File: lib/tasks/get_zodiacs.rake [hunk 1]
```diff
@@ -0,0 +1,27 @@
+namespace :get_zodiacs do
```
Comment (h8rry) [inline lib/tasks/get_zodiacs.rake]: Do you still need this file now that you are scraping from the controller??

File: lib/tasks/scrape_horo.rake [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    data_signs = "div > div > div:nth-child(13) > p:nth-child(1).c3 > a"
+    signs = html_doc.css(data_signs)
+    signs.each do |sign|
+      scrapePage(url + sign.attr('href'), sign.text)
+    end
+  end
+
+  def scrapePage(url, starSign) 
+   begin
```
Comment (h8rry) [inline lib/tasks/scrape_horo.rake]: indentation problem

File: app/assets/javascripts/angular/controller/home-controller.js.erb [hunk 3]
```diff
@@ -5,9 +5,24 @@ app.controller('HomeCtrl', ['$scope', '$http', '$rootScope', function($scope, $h
...
+  
+
+  $scope.clear = function() {
+    $scope.card.date = null;
+  };
+  $scope.minDate = new Date("1946-01-01");
+  $scope.maxDate = new Date();
+  $scope.open = function($event) {
+    $event.preventDefault();
+    $event.stopPropagation();
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/home-controller.js.erb:14]: Why do you need this?

File: app/assets/javascripts/angular/controller/home-controller.js.erb [hunk 4]
```diff
@@ -22,9 +37,16 @@ app.controller('HomeCtrl', ['$scope', '$http', '$rootScope', function($scope, $h
       $scope.card.bestActressMovie = data.movie[0]["bestActressMovie"];
     })
     $http.get('songlists/' + year + '.json').success(function(data){
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/home-controller.js.erb]: Maybe you want to make this a separate function just like how you did it with `getHoroscopes`

File: app/assets/javascripts/angular/controller/home-controller.js.erb [hunk 5]
```diff
@@ -44,10 +66,15 @@ app.controller('HomeCtrl', ['$scope', '$http', '$rootScope', function($scope, $h
 
   $scope.$container.find('.item').each( function( i, itemElem ) {
     // make element draggable with Draggabilly
-    var draggie = new Draggabilly( itemElem );
+    // var draggie = new Draggabilly( itemElem );
     // bind Draggabilly events to Packery
-    $scope.$container.packery( 'bindDraggabillyEvents', draggie );
+    // $scope.$container.packery( 'bindDraggabillyEvents', draggie );
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/home-controller.js.erb:77]: You don't need these anymore?

File: app/assets/javascripts/static_pages.js.coffee [hunk 6]
```diff
@@ -1,3 +0,0 @@
-# Place all the behaviors and hooks related to the matching controller here.
```
Comment (h8rry) [inline app/assets/javascripts/static_pages.js.coffee:1]: can delete the file if you don't need it

File: app/controllers/chinese_horoscopes_controller.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+class ChineseHoroscopesController < ApplicationController
+end
```
Comment (h8rry) [inline app/controllers/chinese_horoscopes_controller.rb:2]: not using yet?

File: lib/tasks/get_zodiacs.rake (hunk 1 above)
Comment (gduck) [inline lib/tasks/get_zodiacs.rake]: No you are right. It's gone

File: lib/tasks/scrape_horo.rake (hunk 2 above)
Comment (gduck) [inline lib/tasks/scrape_horo.rake]: 1 space! ok I fixed it

File: app/assets/javascripts/angular/controller/home-controller.js.erb (hunk 3 above)
Comment (gduck) [inline app/assets/javascripts/angular/controller/home-controller.js.erb:14]: yes you are right. I'm not using the clear function - left over from playing with the datepicker. The max & min are being used though, and open is necessary for the popup window

File: app/assets/javascripts/angular/controller/home-controller.js.erb (hunk 4 above)
Comment (gduck) [inline app/assets/javascripts/angular/controller/home-controller.js.erb]: ok

File: app/assets/javascripts/angular/controller/home-controller.js.erb (hunk 5 above)
Comment (gduck) [inline app/assets/javascripts/angular/controller/home-controller.js.erb:77]: there are layout issues I need to tackle next week. I may use the dragabilly stuff, I'm not sure yet. But for now it's distracting so I'm turning it off

File: app/assets/javascripts/static_pages.js.coffee (hunk 6 above)
Comment (gduck) [inline app/assets/javascripts/static_pages.js.coffee:1]: its' gone

File: app/controllers/chinese_horoscopes_controller.rb (hunk 7 above)
Comment (gduck) [inline app/controllers/chinese_horoscopes_controller.rb:2]: not yet...

**Pull request merged**